dibs/models/linearGaussian.py

```diff
@@ -362,19 +347,6 @@ def log_likelihood(self, *, x, theta, g, interv_targets):
     Distributions used by DiBS for inference:  prior and joint likelihood 
     """
 
-    def log_graph_prior(self, g_prob):
```

Comment:
I would like to leave these functions explictly there. That way, each `inference_model` class implements two functions: `log_graph_prior` and `observational_log_joint_prob`. See in the docstring of `JointDiBS` (same for `MarignalDiBS`). 
My reasoning is that, this way, it will be more modular to define new model classes (only needing to implement 2 functions), without requiring to have an explicit `graph_dist.unnormalized_log_prob_soft` property defined inside the class.

Reply:
Actually, if you're willing to make backwards-incompatible changes, I would take the `graph_dist` entirely out of the inference model: it is _only_ used for the `log_graph_prior`. Nothing else within the inference models depends on it. So removing it would make it more modular: you just have to pass the DiBS "parent model" both the graph model and the inference model (for the functions), instead of inference-model(graph-model)... such separation of concerns would in the longer run also make it cleaner to write tests (being able to instantiate an inference model without having to pass it a "dummy" graph model), for example. But I'm also happy to take this change out of the current PR, if you prefer:)

Reply:
I would just in general like to see the various parts of the code have less arguments that need to keep getting passed around... some thoughts: for example, a "Config" class that bundles the relevant hyperparameters (that you might want to adjust for new datasets) in a single container (`@dataclass`).

Reply:
Another one: `DiBS.particle_to_g_lim` and `DiBS.sample_g` do not actually depend on the `self` instance and could be `@staticmethod`s. Or functions of their own. But also I'm wondering if it would help to encapsulate all the "graph representation" functionality (how to go from $Z$ to edge probabilities / log-probabilities / Bernoulli-sampled graph / hard-thresholded graph) in one class (might be easiest to simply pass alpha_t instead of having it include the alpha function and being passed t itself)?

Reply:
I'll move it to a different PR, just to separate the trivial changes from something breaking backwards compatibility.

Reply:
regarding hyperparameters, I was thinking of this because the paper says that you have to adapt them based on the setting, and they also differ between Joint and Marginal models, so it would make it easier for users to have everything they _should_ touch in one place...

Reply:
That's true, though I think the default hyperparameters of MarginalDiBS and JointDiBS reflect that and tend to work well out of the box. There is no place other than at instantiation when one needs to specify the hyperparameters, so currently I don't see a need for a more elaborate solution, maybe something to consider in the future